src/aspire/basis/polar_2d.py

```diff
@@ -116,11 +114,21 @@ def evaluate_t(self, x):
         x, sz_roll = unroll_dim(x, self.ndim + 1)
         nimgs = x.shape[2]
 
-        # finufftpy require it to be aligned in fortran order
-        half_size = self.nrad * self.ntheta // 2
-        pf = np.empty((half_size, nimgs), dtype='complex128', order='F')
-        finufftpy.nufft2d2many(self.freqs[0, :half_size], self.freqs[1, :half_size], pf, 1, 1e-15, x)
-        pf = m_reshape(pf, (self.nrad, self.ntheta // 2, nimgs))
+        half_size = self.ntheta // 2
+
+        if self.dtype == np.float32:
```

Comment:
Is there a more canonical way to do this (in Numpy)? If not, it might make sense to have util functions (`real_dtype` and `complex_dtype`?) that give the corresponding dtypes.

Reply:
I didn't it from Numpy and will add util functions.

Reply:
In the new commit 5959df6, i added two util functions to match numpy complex type with  real type in Numpy. The self.dtype will be set to the real type from the input x so that there should be no conflicts when calling evaluate function for v and it will return the same type as original x.